src/test/java/no/unit/nva/cristin/projects/CristinApiClientTest.java

```diff
@@ -48,4 +61,57 @@ void getsCristinUriWithProjectCodeParamWhenCallingUriBuilderWithGrantIdQueryRequ
 
         assertEquals(expectedUri, cristinApiClient.generateQueryProjectsUrl(params, QUERY_USING_GRANT_ID));
     }
+
+    @Test
+    void usesQueryResultWhenEnrichingFails() {
+        List<CristinProject> queryProjects = getSomeCristinProjects();
+        List<CristinProject> enrichedProjects = new ArrayList<>(queryProjects);
+        enrichedProjects.remove(enrichedProjects.size() - 1);
+
+        assertThat(enrichedProjects.size(), not(queryProjects.size()));
+        assertThat(idsMatchesOriginalSet(enrichedProjects), equalTo(false));
+
+        List<CristinProject> combinedProjects =
+            cristinApiClient.combineResultsWithQueryInCaseEnrichmentFails(queryProjects, enrichedProjects);
+
+        assertThat(combinedProjects.size(), equalTo(queryProjects.size()));
+        assertThat(idsMatchesOriginalSet(combinedProjects), equalTo(true));
+    }
+
+    @Test
+    void isSuccessfulRequestReturnsCorrectEvaluationWhenSuppliedWithBothValidResponseAndInvalidResponse()
```

Comment:
I think this test is too close to the implementation. I think the original test would be that 
"the client returns both enriched and non-enriched projects when some enrichment queries fail" or something like this.